include/revng/Support/IntegerSerialization.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/Support/Endian.h"
+
+#include "revng/ADT/CompilationTime.h"
+#include "revng/ADT/Concepts.h"
+
+namespace detail {
+
+inline constexpr auto IntEndianness = llvm::support::little;
+inline constexpr auto Unaligned = llvm::support::unaligned;
+
+template<SpecializationOf<std::tuple> T>
+inline constexpr size_t computeSize() {
+  size_t Result = 0;
+  compile_time::repeat<std::tuple_size_v<T>>([&Result]<size_t I>() {
+    Result += sizeof(std::tuple_element_t<I, T>);
+  });
+  return Result;
+}
+
+template<typename... T>
+struct SerializedIntsSize {};
+
+template<isUintOrEnum... Args>
+struct SerializedIntsSize<Args...> {
+  static constexpr size_t Size = computeSize<std::tuple<Args...>>();
+};
+
+template<SpecializationOf<std::tuple> T>
+struct SerializedIntsSize<T> {
+  static constexpr size_t Size = computeSize<T>();
+};
+
+template<isUintOrEnum T>
+void writeNext(uint8_t *&Ptr, const T &Value) {
+  llvm::support::endian::write<T, Unaligned>(Ptr, Value, IntEndianness);
+  Ptr += sizeof(T);
+}
+
+template<isUintOrEnum T>
+T readNext(const uint8_t *&Ptr) {
+  return llvm::support::endian::readNext<T, Unaligned>(Ptr, IntEndianness);
+}
```

Comment:
The fact that `writeNext` increments `Ptr` but `readNext` doesn't is veeeery suspicious.

Reply:
OK, but add a comment about this fact.